internal/jobqueue/fsjobqueue/fsjobqueue.go

```diff
@@ -40,13 +40,18 @@ type fsJobQueue struct {
 	// Maps job ids to the jobs that depend on it, if any of those
 	// dependants have not yet finished.
 	dependants map[uuid.UUID][]uuid.UUID
+
+	// maps token to jobID
+	mapMu        sync.RWMutex
```

Comment:
Ah yea I think I can repurpose `mu`. I used a new mutex, but a `RWMutex` to have a better concurrent reading/writing of maps, but we can just reuse `mu` as this will probably not be the bottleneck :D

Reply:
And if it is, probably the db is the solution ;)